src/schemas/markupmapping.description.md

```diff
@@ -5,5 +5,228 @@ A configuration consits of a mandatory `match` expression, which is a CSS select
 
 - `wrap`: adds more HTML elements around the generated HTML
-- `class` adds `class` attribute values into the generated HTML element
-- `attribute` adds other attributes and values into the generated HTML element
\ No newline at end of file
+- `classnames` adds `class` attribute values into the generated HTML element
+- `attribute` adds other attributes and values into the generated HTML element
+
+## Examples
+
+### On the DOM tree
+
+The most intuitive usage for most developers will be directly operating on the resulting DOM tree.
+The `match` method just takes a regular CSS selector and then applies the desired markup on the resulting element.
+
+#### Adding a class
+
+`index.md`
+```markdown
+# Lorem ipsum dolor sit amet
+
+consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
+
+![Lorem ipsum](http://dolor.sit/amet.jpeg)
+```
+
+`helix-markup.yaml`
+```yaml
+version: 1
+markup:
+  foo:
+    match: p
+    classnames:
+      - foo
+    type: html
+```
+
+`index.html`
+```html
+<h1>Lorem ipsum dolor sit amet</h1>
+<p class="foo">consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua</p>
+<img src="http://dolor.sit/amet.jpeg" alt="Lorem ipsum"/>
+```
+
+#### Adding an attribute
+
+`index.md`
+```markdown
+# Lorem ipsum dolor sit amet
+
+consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
+
+![Lorem ipsum](http://dolor.sit/amet.jpeg)
+```
+
+`helix-markup.yaml`
+```yaml
+version: 1
+markup:
+  foo:
+    match: p
+    attribute:
+      bar: baz
+    type: html
+```
+
+`index.html`
+```html
+<h1>Lorem ipsum dolor sit amet</h1>
+<p bar="baz">consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua</p>
+<img src="http://dolor.sit/amet.jpeg" alt="Lorem ipsum"/>
+```
+
+#### Wrapping with another element
+
+`index.md`
+```markdown
+# Lorem ipsum dolor sit amet
+
+consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
+
+![Lorem ipsum](http://dolor.sit/amet.jpeg)
+```
+
+`helix-markup.yaml`
+```yaml
+version: 1
+markup:
+  foo:
+    match: p
+    wrap: section.qux
+    type: html
+```
+
+`index.html`
+```html
+<h1>Lorem ipsum dolor sit amet</h1>
+<section class="qux">
+  <p>consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua</p>
+</section>
+<img src="http://dolor.sit/amet.jpeg" alt="Lorem ipsum"/>
+```
+
+### On the Markdown abstract tree
+
+As an alternatie, it is also possible to directly operate on the Markdown abstract tree before it is converted to HTML. This is especially useful if you are heavily modifying the markup via the `*.pre.js` and need to already annotate your tree before processing. The `match` property takes a CSS like selector, but instead of DOM elements, you are targeting [MDAST nodes](https://github.com/syntax-tree/mdast#nodes).
```

Comment:
```suggestion
As an alternative, it is also possible to directly operate on the Markdown abstract syntax tree before it is converted to HTML. This is especially useful if you are heavily modifying the markup via the `*.pre.js` and need to already annotate your tree before processing. The `match` property takes a CSS like selector, but instead of DOM elements, you are targeting [MDAST nodes](https://github.com/syntax-tree/mdast#nodes).
```